### PR TITLE
Adding StashCache docs.

### PR DESCRIPTION
**Pull request by djw8605**

### CODE REVIEW

File: docs/data/stashcache/upgrades.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# OSG StashCache Upgrade Status
```
Comment (brianhlin) [inline docs/data/stashcache/upgrades.md]: I think this page belongs in the ops repo

File: docs/data/stashcache/install.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* [StashCache origin](configure-origin.md): **"origin"** is the authoritative data server that hosts files locally and serves them to users upon transfer request (via [stashcp](https://support.opensciencegrid.org/support/solutions/articles/12000002775-transferring-data-with-stashcach) or CVMFS) unless data are already cached.
+
+---
+
+## Installation prerequisites for Origin and Cache
+
+Before starting the installation process, consider the following mandatory points:
+
+* __User IDs:__ If they do not exist already, the installation will create the Linux user IDs condor and xrootd
+* __Service certificate:__ The StashCache server uses a host certificate to advertise to a central collector.  More information on how to receive a certificate can be found [here](/security/host-certs.md)
```
Comment (brianhlin) [inline docs/data/stashcache/install.md]: Let's call this bullet "Host certificate"

Comment (brianhlin) [inline docs/data/stashcache/install.md]: Let's call this bullet "Host certificate"

**Comment by brianhlin** [comment]
D'oh, I didn't mean to submit the review just yet. The split of the pages is confusing, I think it should be, at most:

```
StashCache\
    Overview
    Installing an Origin Server
    Installing a Cache Server
```

i.e. join the install and config pages based on the host. Maybe we don't even need a separate overview page.

Who generally installs an Origin server? VOs?

File: docs/data/stashcache/upgrades.md (hunk 1 above)
Comment (djw8605) [reply docs/data/stashcache/upgrades.md]: I would agree.  Though, one of the most common questions asked to the StashCache team is: "What are the specs of the caches?"

**Comment by djw8605** [comment]
The installation is almost the same, it's the configuration that is different.  If we split to "Install Origin", and "Install Cache", everything before the configure parts will be duplicates.  I suppose that is fine with me, just need to remember both install instructions if package names change, or any changes in the installation procedure.

But other documents can be merged.  The `start.md` should be merged into the configure.  The optional cache configuration and auth configuration can be merged into the cache configuration.

File: docs/data/stashcache/upgrades.md (hunk 1 above)
Comment (brianhlin) [reply docs/data/stashcache/upgrades.md]: Then let's link it from the cache install page

**Comment by osg-cat** [comment]
As a new user, I think I would ask (a) what StashCache is, and then (b) what parts I needed given my situation and what value it/they will add. Is part (b) addressed anywhere?

File: docs/data/stashcache/overview.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Overview of StashCache
+
+StashCache is a data caching federation with many distributed caches spread across the U.S.
```
Comment (brianhlin) [inline docs/data/stashcache/overview.md]: This describes what StashCache is but doesn't really describe who the target audience is. Maybe word it something like this, expanding on `(for some value of large)` either here or below

> StashCache is a data caching federation used by large VOs (for some value of large) and the sites that support them.  The federation has many distributed caches spread across the U.S.

File: docs/data/stashcache/overview.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+
+![StashCache Map](StashCacheMap.png "StashCache Map")
+
+The StashCache federation consists of two service types, the Caches and the Origins:
+
+* StashCache Cache: role of **"cache"** server is to keep data cached and immediately available (via [stashcp](https://support.opensciencegrid.org/support/solutions/articles/12000002775-transferring-data-with-stashcach) or CVMFS) within Stash federation (without re-transferring from "origin").
```
Comment (brianhlin) [inline docs/data/stashcache/overview.md]: Presumably not all admins should install caches. What conditions would prompt an admin to set up a cache?

File: docs/data/stashcache/overview.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+* StashCache Origin: Authoritative copy data available to the caching servers.  The origin is only accessed when the Caches do not have the requested data.
```
Comment (brianhlin) [inline docs/data/stashcache/overview.md]: Do VOs set up their own origins or are they all OSG maintained?

File: docs/data/stashcache/install-origin.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+* StashCache cache: role of **"cache"** server is to keep data cached and immediately available (via [stashcp](https://support.opensciencegrid.org/support/solutions/articles/12000002775-transferring-data-with-stashcach) or CVMFS) within Stash federation (without re-transferring from "origin").
+* StashCache origin: The authoritative data server that hosts files locally and serves them to users upon transfer request (via [stashcp](https://support.opensciencegrid.org/support/solutions/articles/12000002775-transferring-data-with-stashcach) or CVMFS) unless data are already cached.
+
+---
+
+## Installation prerequisites for Origin
+
+Before starting the installation process, consider the following mandatory points:
+
+* __User IDs:__ If they do not exist already, the installation will create the Linux user IDs condor and xrootd
```
Comment (brianhlin) [inline docs/data/stashcache/install-origin.md]: What are the numeric UIDs?

File: docs/data/stashcache/install-origin.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+* __Host certificate:__ The StashCache server uses a host certificate to advertise to a central collector.  More information on how to receive a certificate can be found [here](/security/host-certs.md)
```
Comment (brianhlin) [inline docs/data/stashcache/install-origin.md]: "retrieve" probably makes more sense than "receive" here

File: docs/data/stashcache/install-origin.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+* __Network ports:__ The StashCache service must listen on ports:
+    * XRootD service on port `1094 (TCP)`
```
Comment (brianhlin) [inline docs/data/stashcache/install-origin.md]: Why make this a sub-bullet if there's only one port?

File: docs/data/stashcache/install-origin.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+* Ensure the host has [a supported operating system](/release/supported_platforms.md)
+* Obtain root access to the host
+* Prepare [the required Yum repositories](/common/yum.md)
+* Install [CA certificates](/common/ca.md)
+
+## Installing the StashCache metapackage
+
+The StashCache daemon consists of an XRootD server and an HTCondor-based service for collecting and reporting statistics about the cache. To simplify installation, OSG provides convenience RPMs that install all required software with a single command:
+
+    [root@client ~]$ yum install stashcache-daemon fetch-crl stashcache-cache-origin
```
Comment (brianhlin) [inline docs/data/stashcache/install-origin.md]: See these sections: 

https://opensciencegrid.org/technology/documentation/style-guide/#command-blocks-and-file-snippets
https://opensciencegrid.org/technology/documentation/style-guide/#root-and-user-prompts

File: docs/data/stashcache/install-origin.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+   
+
+Mount the disk that will be used for the origin data to */stash* and set owner of the directory to `xrootd:xrootd` user.  
+
+
+# Configuring Origin Server
+
+Packages installed: `stashcache-daemon fetch-crl stashcache-origin-server`
```
Comment (brianhlin) [inline docs/data/stashcache/install-origin.md]: We just told them to install these packages, drop this line

File: docs/data/stashcache/install-origin.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+The following section describes required configuration to have a functional StashCache Origin (not cache server!). StashCache Origin package `stashcache-cache-origin` needs to be manually configured from pre-existing XRootD configuration.
```
Comment (brianhlin) [inline docs/data/stashcache/install-origin.md]: The paranthetical is repetitive. The grammar of the second sentence is suspect

File: docs/data/stashcache/overview.md (hunk 3 above)
Comment (djw8605) [reply docs/data/stashcache/overview.md]: Added comment about who uses StashCache.  It's actually the small VO's.  No big VO uses StashCache.

File: docs/data/stashcache/overview.md (hunk 4 above)
Comment (djw8605) [reply docs/data/stashcache/overview.md]: Not all should, probably.  But any can.  Changed.

File: docs/data/stashcache/overview.md (hunk 5 above)
Comment (djw8605) [reply docs/data/stashcache/overview.md]: VO's.  Only 1 is OSG maintained, the one at UChicago.

File: docs/data/stashcache/install-origin.md (hunk 6 above)
Comment (djw8605) [reply docs/data/stashcache/install-origin.md]: No idea.  Does that matter?

File: docs/data/stashcache/overview.md (hunk 3 above)
Comment (brianhlin) [reply docs/data/stashcache/overview.md]: Apart from the issue with the grammar in the sentence saying that stashcache is for researchers and sites isn't really informative: all of our external-facing software supports researchers or sites in some capacity.

If it's used by small VOs, what would prompt a small VO to set up stashcache origins/caches?

File: docs/data/stashcache/install-origin.md (hunk 6 above)
Comment (brianhlin) [reply docs/data/stashcache/install-origin.md]: I don't think so but it's information we try to provide in other installation docs

File: docs/data/stashcache/overview.md [hunk 12]
```diff
@@ -7,8 +7,8 @@ StashCache is a data caching federation used researchers and the sites that supp
...
-* StashCache Origin: Authoritative copy data available to the caching servers.  The origin is only accessed when the Caches do not have the requested data.
+* StashCache Cache: role of **"cache"** server is to keep data cached and immediately available (via [stashcp](https://support.opensciencegrid.org/support/solutions/articles/12000002775-transferring-data-with-stashcach) or CVMFS) within Stash federation (without re-transferring from "origin").  StashCache currently consists of multiple regional caches, but any site may install their own cache and make it available to either only their site, or the wider user community.  A cache can reduce the amount of data transferred through the wide area network if you have OSG users that use StashCache data distribution.
+* StashCache Origin: Authoritative copy data available to the caching servers.  The origin is only accessed when the Caches do not have the requested data.  The Origin is run and maintained by organizations distributing data within the StashCache federation.
```
Comment (brianhlin) [inline docs/data/stashcache/overview.md]: Could we change "distributing" to "who wish to distribute"?

**Pull request merged**